src/api/procedures/renamePortfolio.ts

```diff
@@ -34,35 +39,26 @@ export async function prepareRenamePortfolio(
   const { did, id, name: newName } = args;
 
   const identityId = stringToIdentityId(did, context);
-  const rawPortfolioNumber = numberToU64(id, context);
 
-  const [rawPortfolioName, rawPortfolios] = await Promise.all([
-    queryPortfolio.portfolios(identityId, rawPortfolioNumber),
-    queryPortfolio.portfolios.entries(identityId),
-  ]);
+  const rawNewName = stringToText(newName, context);
 
-  if (textToString(rawPortfolioName) === newName) {
-    throw new PolymeshError({
-      code: ErrorCode.NoDataChange,
-      message: 'New name is the same as current name',
-    });
-  }
-
-  const portfolioNames = rawPortfolios.map(([, name]) => textToString(name));
+  const existingPortfolioNumber = await portfolioNameToNumber(identityId, rawNewName, query);
 
-  if (portfolioNames.includes(newName)) {
-    throw new PolymeshError({
-      code: ErrorCode.UnmetPrerequisite,
-      message: 'A portfolio with that name already exists',
-    });
+  if (existingPortfolioNumber) {
+    if (id.eq(existingPortfolioNumber)) {
+      throw new PolymeshError({
+        code: ErrorCode.NoDataChange,
+        message: 'New name is the same as current name',
+      });
+    } else {
+      throw new PolymeshError({
+        code: ErrorCode.UnmetPrerequisite,
+        message: 'A portfolio with that name already exists',
```

Comment:
We aren't consistent, but I think we want entities to be capitalized like the rest API, so:

```suggestion
        message: 'A Portfolio with that name already exists',
```